snorkel/learning/gen_learning.py

```diff
@@ -169,13 +74,13 @@ class GenerativeModel(object):
 
     :param class_prior: whether to include class label prior factors
     :param lf_prior: whether to include labeling function prior factors
-    :param lf_propensity: whether to include labeling function propensity 
+    :param lf_propensity: whether to include labeling function propensity
         factors
-    :param lf_class_propensity: whether to include class-specific labeling 
+    :param lf_class_propensity: whether to include class-specific labeling
         function propensity factors
     :param seed: seed for initializing state of Numbskull variables
     """
-    def __init__(self, class_prior=False, lf_prior=False, lf_propensity=False, 
+    def __init__(self, class_prior=False, lf_prior=False, lf_propensity=False,
```

Comment:
My understanding is that if you run this code against an old version of Numbskull with binary classes, it will silently do the wrong thing. That seems *very bad*.

I don't think we can cleanly handle the new NS / old Snorkel case, but can we at least raise an error in the new Snorkel / old NS case by checking for some flag set in the NS package?